python/lsst/analysis/ap/apdb.py

```diff
@@ -81,21 +145,33 @@ def load_sources_for_object(self, dia_object_id, limit=100000):
         data : `pandas.DataFrame`
             DiaSources for the specified diaObject.
         """
+
+        where_clauses = [f'"DiaSource"."diaObjectId" = {dia_object_id}']
+
+        if exclude_flagged:
+            where_clauses.append(self._make_flag_exclusion_clause(self.bad_DiaSource_flags))
+
+        where = "WHERE " + " and ".join(where_clauses) if len(where_clauses) else ""
+
+        order = 'ORDER BY "ccdVisitId", "diaSourceId"'
+        limit_str = f"LIMIT {limit}" if limit is not None else ""
+        query = ('SELECT * FROM "DiaSource"'
+                 f' {where} {order} {limit_str};')
+
         with self.connection as connection:
-            order = 'ORDER BY "ccdVisitId", "diaSourceId"'
-            limit_str = f"LIMIT {limit}" if limit is not None else ""
-            query = ('SELECT * FROM "DiaSource"'
-                     f' WHERE "DiaSource"."diaObjectId" = {dia_object_id}'
-                     f' {order} {limit_str};')
             result = pd.read_sql_query(query, connection)
+
         self._fill_from_ccdVisitId(result)
         return result
 
-    def load_sources(self, limit=100000):
-        """Load all DiaSources.
+    def load_sources(self, exclude_flagged=False, limit=100000):
+        """Load DiaSources.
 
         Parameters
         ----------
+        exclude_flagged : `bool`, optional
+            If True, do not return records with configurable flags
```

Comment:
Perhaps: "Exclude sources that have flags set (see `set_bad_DiaSource_flags` to configure those flags)"?